tests/Phoenix.MarketData.Infrastructure.Tests/generate-coverage-report.sh

```
@@ -0,0 +1,85 @@
+# Check if test results directory exists
+if [ ! -d "$TEST_RESULTS_DIR" ]; then
+    echo "Error: Test results directory not found: $TEST_RESULTS_DIR"
+    echo "Tests may have run without generating coverage data."
+    exit 1
+fi
+
+# Create the report directory if it doesn't exist
+mkdir -p "$REPORT_DIR"
+
+# Find all coverage files
+COVERAGE_FILES=( $(find "$TEST_RESULTS_DIR" -name "coverage.cobertura.xml" 2>/dev/null) )
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)